### PR TITLE
make nitunit  more colorful (and useful?)

### PR DESCRIPTION
**Pull request by privat**
This series move more information in the UnitTest class, this allow specific and unified way of displaying the test results.

See the asciinema: https://asciinema.org/a/108jd2a3tkfifg3wzoacg0a65

Comments? Needs? Ideas?

### CODE REVIEW

File: src/testing/testing_base.nit [hunk 1]
```diff
@@ -93,41 +94,127 @@ ulimit -t {{{ulimit_usertime}}} 2> /dev/null
...
+		line += "] {done}/{tests.length}"
+		if more_message != null then
+			line += " " + more_message
+		end
+		printn "{line}"
+	end
+
 end
 
 # A unit test is an elementary test discovered, run and reported bu nitunit.
```
Comment (Morriar) [inline src/testing/testing_base.nit]: by*

File: src/testing/testing_base.nit [hunk 2]
```diff
@@ -93,41 +94,127 @@ ulimit -t {{{ulimit_usertime}}} 2> /dev/null
...
 	var was_exec = false is writable
 
-	# Return the `TestCase` in XML format compatible with Jenkins.
+	# The raw output of the execution (or compilation)
 	#
-	# See to_xml
+	# It merges the standard output and error output
+	var raw_output: nullable String = null is writable
+
+	# The location where the error occurred, if it make sense.
```
Comment (Morriar) [inline src/testing/testing_base.nit]: makes*

File: src/testing/testing_base.nit [hunk 3]
```diff
@@ -93,41 +94,127 @@ ulimit -t {{{ulimit_usertime}}} 2> /dev/null
...
+	var error_location: nullable Location = null is writable
+
+	# A colorful `[OK]` or `[KO]`.
+	fun status_tag: String
+	do
```
Comment (Morriar) [inline src/testing/testing_base.nit]: You should use inlined do to preserve a coherent style in the module :D

File: src/testing/testing_base.nit [hunk 4]
```diff
@@ -93,41 +94,127 @@ ulimit -t {{{ulimit_usertime}}} 2> /dev/null
...
+			end
+		else
+			res = "{status_tag} {full_name}"
+			if more_message != null then res += more_message
+		end
+		return res
+	end
+
+	# Return a `<testcase>` XML node in format compatible with Jenkins unit tests.
 	fun to_xml: HTMLTag do
```
Comment (Morriar) [inline src/testing/testing_base.nit]: Yes, they all need to be coherent with this one.

**Comment by Morriar** [comment]
+1

**Comment by privat** [comment]
reroll apply remarks of @Morriar

**Comment by jcbrinfo** [comment]
:+1: +1, but you should add a `--no-color` option for non-standard terminals and redirections to a file.

**Comment by privat** [comment]
Yeah, I will respect --no-color in a future PR

**Pull request merged**